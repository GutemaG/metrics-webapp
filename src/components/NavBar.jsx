import React from 'react';
import { Button } from 'react-bootstrap';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import { Link, useLocation, useNavigate } from 'react-router-dom';
import './style/NavBar.css';

const NavBar = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const goBack = () => {
    navigate(-1);
  };

  return (
    <Navbar variant="dark" bg="dark" expand="lg" className="nav-bar">
      <Container>
        <Link to="/" style={{ color: '#fff' }}>CryptoCurrencies</Link>
        <Navbar.Toggle aria-controls="navbarScroll" />
        <Navbar.Collapse id="navbarScroll">
          <Nav
            className="me-auto my-2 my-lg-0"
            style={{ maxHeight: '100px' }}
            navbarScroll
          >
            {
              !(location.pathname === '/') && (
                <Button
                  variant="outline-secondary"
                  className="go-back-arrow"
                  onClick={goBack}
                >
                  &larr;
                </Button>
              )
            }
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default NavBar;
