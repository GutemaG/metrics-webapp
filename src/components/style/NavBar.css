.go-back-arrow {
  width: 40px;
  height: 40px;
  color: #fff;
  border-radius: 50%;
  margin-left: 15px;
}

a {
  color: red;
  text-decoration: none;
}
