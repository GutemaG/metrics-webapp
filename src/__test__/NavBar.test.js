import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import NavBar from '../components/NavBar';

describe('Testing NavBar page', () => {
  test('Render NavBar page correctly', () => {
    const navbar = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>,
    );
    expect(navbar).toMatchSnapshot();
    expect(navbar.getByText('CryptoCurrencies')).toBeInTheDocument();
  });
});
