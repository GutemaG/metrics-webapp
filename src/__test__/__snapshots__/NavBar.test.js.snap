// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NavBar page Render NavBar page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav-bar navbar navbar-expand-lg navbar-dark bg-dark"
      >
        <div
          class="container"
        >
          <a
            href="/"
            style="color: rgb(255, 255, 255);"
          >
            CryptoCurrencies
          </a>
          <button
            aria-controls="navbarScroll"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="navbarScroll"
          >
            <div
              class="me-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
              style="max-height: 100px;"
            />
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav-bar navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        class="container"
      >
        <a
          href="/"
          style="color: rgb(255, 255, 255);"
        >
          CryptoCurrencies
        </a>
        <button
          aria-controls="navbarScroll"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="navbarScroll"
        >
          <div
            class="me-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
            style="max-height: 100px;"
          />
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
